plugins/source/gcp/resources/services/networkconnectivity/locations.go

```diff
@@ -0,0 +1,52 @@
+package networkconnectivity
+
+import (
+	"context"
+
+	"github.com/apache/arrow/go/v14/arrow"
+	"google.golang.org/api/networkconnectivity/v1"
+
+	"github.com/cloudquery/cloudquery/plugins/source/gcp/client"
+	"github.com/cloudquery/plugin-sdk/v4/schema"
+	"github.com/cloudquery/plugin-sdk/v4/transformers"
+)
+
+func Locations() *schema.Table {
+	return &schema.Table{
+		Name:        "gcp_networkconnectivity_locations",
+		Description: `https://cloud.google.com/network-connectivity/docs/reference/networkconnectivity/rest/v1/projects.locations/list`,
+		Resolver:    fetchLocations,
+		Multiplex:   client.ProjectMultiplexEnabledServices("networkconnectivity.googleapis.com"),
+		Transform:   client.TransformWithStruct(&networkconnectivity.Location{}, transformers.WithPrimaryKeys("Name")),
+		Columns: []schema.Column{
+			{
+				Name:       "project_id",
+				Type:       arrow.BinaryTypes.String,
+				Resolver:   client.ResolveProject,
+				PrimaryKey: true,
+			},
+		},
```

Comment:
It may be a good idea to reuse code here:
https://github.com/cloudquery/cloudquery/blob/main/plugins/source/gcp/client/columns.go#L8
```suggestion
		Columns: schema.ColumnList{client.ProjectIDColumn(true)},
```

We tend to add the `project_id` as PK for locations, see [`gcp_kms_locations`](https://www.cloudquery.io/docs/plugins/sources/gcp/tables/gcp_kms_locations).

Reply:
done